### PR TITLE
build: skip qlc modules if there were no git diff

### PR DESCRIPTION
**Pull request by zmstone**
All the listed modules with qlc.hrl included, qlc is the only header.
so check git diff of the module is safe.

https://emqx.atlassian.net/browse/EMQX-8655

### CODE REVIEW

File: scripts/update_appup.escript [hunk 1]
```diff
@@ -66,9 +66,50 @@ ignored_apps() ->
...
+
+is_git_diff(BaseVsn, File) ->
+    CMD = "git diff --name-only "++ BaseVsn ++ "...HEAD " ++ File,
+    case string:trim(os:cmd(CMD)) of
+        "" ->
+            false;
+        File ->
+            true;
+        Other ->
+            logerr("Failed to git diff ~s:~n", [File]),
```
Comment (zmstone) [inline scripts/update_appup.escript:96]:
```suggestion
            logerr("Failed to git diff: ~s~n", [File]),
```

File: scripts/update_appup.escript [hunk 2]
```diff
@@ -620,7 +664,7 @@ diff_app(UpOrDown, App,
             case UpOrDown =:= up of
                 true ->
                     %% only log for the upgrade case because it would be the same result
-                    logerr("Application '~p' contains changes, but its version is not updated. ~s",
+                    logerr("Application '~p' contains changes, but its version is not updated. ~s~n",
```
Comment (thalesmg) [inline scripts/update_appup.escript:667]: nit: maybe add the `~n` to `logerr`'s definition?

File: scripts/update_appup.escript [hunk 3]
```diff
@@ -66,9 +66,50 @@ ignored_apps() ->
...
+                      {Module, #{path => F,
+                                 is_git_diff => is_git_diff(BaseVsn, F)
+                                }}
+              end, qlc_modules()).
+
 main(Args) ->
     #{prev_tag := Baseline} = Options = parse_args(Args, default_options()),
-    init_globals(Options),
+    QlcModules = qlc_module_git_diff(Baseline),
+    log("Changedd modules which have qlc.hrl included: ~p", [QlcModules]),
```
Comment (thalesmg) [inline scripts/update_appup.escript:111]:
```suggestion
    log("Changed modules which have qlc.hrl included: ~p", [QlcModules]),
```

**Review by thalesmg — APPROVED** [review]
maybe we'd need to assert that the listed files with QLC headers indeed always have only that header to guarantee the git diff output to be safe?

**Review by thalesmg — APPROVED** [review]

**Pull request merged**